plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/revisionsList/RevisionListPresenter.java

```diff
@@ -0,0 +1,192 @@
+                                 DialogFactory dialogFactory,
+                                 DtoUnmarshallerFactory dtoUnmarshallerFactory,
+                                 ProjectExplorerPresenter projectExplorer) {
+        this.view = view;
+        this.comparePresenter = comparePresenter;
+        this.dialogFactory = dialogFactory;
+        this.projectExplorer = projectExplorer;
+        this.gitService = service;
+        this.locale = locale;
+        this.appContext = appContext;
+        this.notificationManager = notificationManager;
+        this.dtoUnmarshallerFactory = dtoUnmarshallerFactory;
+        this.workspaceId = appContext.getWorkspaceId();
+
+        this.view.setDelegate(this);
+    }
+
+    /** Open dialog and shows revisions to compare. */
+    public void showRevisions() {
+        project = appContext.getCurrentProject().getRootProject();
+        String path;
+
+        Selection<ResourceBasedNode< ? >> selection = getExplorerSelection();
+
+        if (selection == null || selection.getHeadElement() == null) {
+            path = project.getPath();
+        } else {
+            path = ((HasStorablePath)selection.getHeadElement()).getStorablePath();
+        }
+
+        String pattern = path.replaceFirst(project.getPath(), "");
+        selectedFile = (pattern.startsWith("/")) ? pattern.replaceFirst("/", "") : pattern;
+
+        getRevisions();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onCloseClicked() {
+        view.close();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onCompareClicked() {
+        compare();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onRevisionUnselected() {
+        selectedRevision = null;
+        view.setEnableCompareButton(false);
+        view.setDescription(locale.viewCompareRevisionFullDescriptionEmptyMessage());
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onRevisionSelected(@NotNull Revision revision) {
+        selectedRevision = revision;
+
+        view.setEnableCompareButton(true);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onRevisionDoubleClicked() {
+        compare();
+    }
+
+    /** Get list of revisions. */
+    private void getRevisions() {
+        gitService.log(workspaceId, project, Collections.singletonList(selectedFile), false,
+                       new AsyncRequestCallback<LogResponse>(dtoUnmarshallerFactory.newUnmarshaller(LogResponse.class)) {
+
+                           @Override
+                           protected void onSuccess(LogResponse result) {
+                               view.setRevisions(result.getCommits());
+                               view.showDialog();
+                           }
+
+                           @Override
+                           protected void onFailure(Throwable exception) {
+                               notificationManager.notify(exception.getMessage(), FAIL, false);
```

Comment:
Use `org.eclipse.che.ide.ext.git.client.GitLocalizationConstant#logFailed` as message instead of `exception.getMessage()`